src/test/java/org/broadinstitute/hellbender/utils/nio/PathLineIteratorUnitTest.java

```diff
@@ -0,0 +1,53 @@
+package org.broadinstitute.hellbender.utils.nio;
+
+import com.google.common.jimfs.Configuration;
+import com.google.common.jimfs.Jimfs;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.file.FileSystem;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import org.broadinstitute.hellbender.GATKBaseTest;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class PathLineIteratorUnitTest extends GATKBaseTest {
+
+    final String[] opus = {"Hello world", "What's new?"};
+
+    @Test
+    public void testLineIteratorInForEach() throws IOException {
+        try (FileSystem jimfs = Jimfs.newFileSystem(Configuration.unix())) {
+            final Path path = jimfs.getPath("test.txt");
+            try (BufferedWriter bufferedWriter = Files.newBufferedWriter(path)) {
+                bufferedWriter.write(opus[0] + "\n");
```

Comment:
Why not use instead `String.join("\n", opus)`. It will simplify things here...

Reply:
Yes that's better. Done.